lxd/storage/drivers/drivers_lvm_volumes.go

```diff
@@ -0,0 +1,732 @@
+package drivers
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"os"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+
+	"github.com/lxc/lxd/lxd/migration"
+	"github.com/lxc/lxd/lxd/operations"
+	"github.com/lxc/lxd/lxd/revert"
+	"github.com/lxc/lxd/lxd/rsync"
+	"github.com/lxc/lxd/shared"
+	log "github.com/lxc/lxd/shared/log15"
+)
+
+// GetVolumeUsage returns the disk space used by the volume (this is not currently supported).
+func (d *lvm) GetVolumeUsage(vol Volume) (int64, error) {
+	return 0, ErrNotSupported
```

Comment:
We should be able to use:

```go
    if vol.contentType == ContentTypeFS && shared.IsMountPoint(vol.MountPath()) {
        var stat unix.Statfs_t
        err := unix.Statfs(vol.MountPath(), &stat)
        if err != nil {
            return -1, err
        }

        return int64(stat.Blocks-stat.Bfree) * int64(stat.Bsize), nil
    }
```

Otherwise on thinpool we should be able to get usage data from LVM right?

Reply:
@stgraber yes I can use that, the old driver didn't, I wasn't sure what contract this function has, i.e whether it should always return usage even if instance is stopped etc.

RE the thinpool question, there is a `data_percent` value I can use for a thin volume that combined with the volume's allocated size can given an approximation of usage. 

However should I only use that for non FS content type volumes, otherwise the usage size for the same volume could change dramatically whether the instance is running or not, as the filesystem is likely to allocate blocks from the thinpool and not give them back, even if they are considered "empty" by the filesystem.

Reply:
So I think our users usually care about whether the container is about to run out of disk space or not, which would be the space use inside of it.

I'd say we use the vfs method for filesystem volumes and use the thinpool percentage for block volumes. If the fs volume isn't mounted we return NotSupported so the caller knows we don't have a value in this case.

Reply:
@stgraber ok sounds good